src/Diagram/stateManager/nodeManager/index.ts

```diff
@@ -217,6 +233,53 @@ export class NodeManager {
       this.eventBus.publish(Events.DiagramEvents.RenderRequested);
     }
   };
+  foldUnfoldChildren = () => {
+    const { node, io } = this.state.hover;
+    if (node && !io) {
+      const root = this.state.nodes.find(f => f.id === node.id);
+      const childrenToFold = new Array<Node>();
+      let foundChildren = new Array<Node>();
+      let foundAllChildren = false;
+
+      if (root === undefined) {
+        return;
+      }
+      let newParents = [root];
+      while (!foundAllChildren) {
+        foundChildren = this.findChildren(newParents);
+        newParents = foundChildren;
+        childrenToFold.push(...foundChildren);
+        foundAllChildren = foundChildren.length === 0 ? true : foundAllChildren;
+      } 
+      if (childrenToFold.length > 0) {
+        childrenToFold.forEach(child => {
+          let node = this.state.nodes.find(f => f.id === child.id);
+          node!.hidden = !(node!.hidden);
```

Comment:
`if (node) { node.hidden = !node.hidden };` is safer

Reply:
Corrected